backend/benefit/applications/api/v1/application_views.py

```diff
@@ -405,6 +407,27 @@ def create(self, request, *args, **kwargs):
     def update(self, request, *args, **kwargs):
         return super().update(self._prune_fields(request), *args, **kwargs)
 
+    def destroy(self, request, *args, **kwargs):
+        # Only the applicant can delete an alteration, and only if it hasn't yet been
+        # opened by a handler.
+
+        alteration = self.get_object()
+
+        if not settings.NEXT_PUBLIC_MOCK_FLAG:
+            company = get_company_from_request(request)
+            if company != alteration.application.company:
+                raise PermissionDenied(_("You are not allowed to do this action"))
+
+        if request.user.is_handler():
```

Comment:
You can use rest framework's `permission_classes = [BFIsHandler]` on the viewset level. Or is the API shared with applicant?

Reply:
Sorry vice versa so `BFIsApplicant` in this case.

Reply:
I'd say if they're shared, or plan to share then decouple as applicant and handler viewset. Otherwise role checking becomes swiss cheese if `permission_class` is not used.

Reply:
Both currently use the same API, yeah. Splitting doesn't sound like a bad idea, so I should do that – I'll do so in the following PRs during the handler side implementation. I'll have to take a closer look at the application list where that already exists but I imagine it should be pretty straightforward.